src/character.cpp

```diff
@@ -1183,11 +1184,18 @@ void Character::drop_invalid_inventory()
     }
 
     if( volume_carried() > volume_capacity() ) {
-        for( auto &item_to_drop :
-             inv.remove_randomly_by_volume( volume_carried() - volume_capacity() ) ) {
-            g->m.add_item_or_charges( pos(), item_to_drop );
+        auto items_to_drop = inv.remove_randomly_by_volume( volume_carried() - volume_capacity() );
+        /// @todo For now this only supports dropping into vehicle for the
+        /// player, not other characters, because put_into_vehicle_or_drop
+        /// requires a player as an argument.  This could probably be
+        /// generalized, but would require looking into.
```

Comment:
NPCs are 'players', what other characters are you referring to here?  Things like monsters don't have this ability.

Reply:
Thanks for the clarification; I had an inkling that NPCs were `player`s but I hadn't quite confirmed that was the case.
Having looked into this, it appears that this function (`Character::drop_invalid_inventory`) is only called on `player`s, so I could just move it from `Character` to `player` and the problem would go away.  Any objection?

Reply:
I actually didn't notice what function we were in here, I just glanced over the comment and it didn't seem to make sense.

Looking into it a bit more, this function should live in the `character` class because the member `inv` lives here.  Moreover I don't see a reason why `put_into_vehicle_or_drop` actually needs a player reference and not `*this` other than the fact it's calling `handle_spillable_contents` which asks for a player reference, but can be changed as well for similar reasons.

Reply:
I considered that option.  I thought it would lead down a big rabbit hole, but you're right, it seems straightforward.  So, that's done and pushed.